x-pack/plugins/session_view/public/components/Process/index.tsx

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useEffect, MouseEvent } from 'react';
+import { EuiButton, EuiIcon, useEuiTheme } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { IProcess } from '../../hooks/use_process_tree';
+
+const TREE_INDENT = 32;
+
+interface IProcessDeps {
+  process: IProcess;
+  isSessionLeader?: boolean;
+  depth?: number;
+  onProcessSelected(process: IProcess): void;
+}
+
+/**
+ * Renders a node on the process tree
+ * TODO: as well as sections for tty output, alerts and file redirection.
+ */
+export function Process({
+  process,
+  isSessionLeader = false,
+  depth = 0,
+  onProcessSelected,
+}: IProcessDeps) {
+  const { euiTheme } = useEuiTheme();
+  const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
+
+  useEffect(() => {
+    setChildrenExpanded(isSessionLeader || process.autoExpand);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [process.autoExpand]);
+
+  const darkTextCSS = `
+    color: ${euiTheme.colors.text};
+  `;
+
+  const searchHighlightCSS = `
+    background-color: ${euiTheme.colors.highlight};
+    color: ${euiTheme.colors.text};
+    border-radius: 4px;
+  `;
```

Comment:
not a huge fan of defining styling inside the component, I guess this is because they need a reference to the `euiTheme` object. maybe we can come up with a better pattern

Reply:
Yea me neither. Not sure what the best approach to this is. I do like co-locating styles with the component, but yea. seems messy.